src/zc/lockfile/__init__.py

```diff
@@ -102,3 +103,61 @@ def close(self):
             _unlock_file(self._fp)
             self._fp.close()
             self._fp = None
+
+
+class LockedContext(object):
+    """ContextManager version of LockFile"""
+
+    def __init__(self, name, content_template='{pid}', timeout=float("inf")):
```

Comment:
The standard library is a bit schizophrenic in how it treats `timeout` values that are supposed to be infinite, but I can't think of a place it uses `inf`. Older APIs use `-1`, while newer APIs use `None` to mean infinite. I think using `None` would be a more expected choice and help avoid mistakes if people transition from in-process locking to multi-process locking.

(I also think it'd be great to have an `acquire(blocking=True, timeout=None)` method that returns a value instead of raising an exception as a primitive, but that's neither here nor there.)